apps/s_cb.c

```diff
@@ -1074,7 +1074,7 @@ static void print_raw_cipherlist(SSL *s)
             BIO_puts(bio_err, ":");
         if (c)
             BIO_puts(bio_err, SSL_CIPHER_get_name(c));
-        else if (!memcmp(rlist, scsv_id - num + 3, num))
+        else if (num <= 3 && !memcmp(rlist, scsv_id - num + 3, num))
```

Comment:
If SSL_get0_raw_cipherlist(s, NULL) ever returns anything other than 2 then that looks to me like a big problem, and I'd like to understand why. Have you actually confirmed that this solves the issue? My suspicion is that |num| is actually 2, and the calculation |scsv_id - num| is triggering the "out of bounds for type 'const
unsigned char [3]'" *before* we then pull it back in bounds again by adding the 3. If so you're proposed fix won't solve the problem.

Probably this whole thing can be simplified (in master at least), because this whole dance is dealing with the fact that in SSLv2 ciphers are encoded as 3 bytes, but in SSLv3+ ciphers are encoded as 2 bytes. In master we don't have SSLv2 any more so |num| should never be anything other than 2.